scylla/src/transport/topology.rs

```diff
@@ -61,6 +61,66 @@ pub struct Peer {
     pub rack: Option<String>,
 }
 
+/// An endpoint for a node that the driver is to issue connections to,
+/// possibly after prior address translation.
+#[non_exhaustive] // <- so that we can add more fields in a backwards-compatible way
+#[derive(Clone)]
+pub enum UntranslatedEndpoint {
+    /// Provided by user in SessionConfig (initial contact points).
+    ContactPoint(ContactPoint),
+    /// Fetched in Metadata with `query_peers()`
+    Peer(PeerEndpoint),
+}
+
+impl UntranslatedEndpoint {
+    pub(crate) fn address(&self) -> SocketAddr {
+        match *self {
+            UntranslatedEndpoint::ContactPoint(ContactPoint { address }) => address,
+            UntranslatedEndpoint::Peer(PeerEndpoint { address, .. }) => address,
+        }
+    }
+    pub(crate) fn address_mut(&mut self) -> &mut SocketAddr {
+        match self {
+            UntranslatedEndpoint::ContactPoint(ContactPoint { address }) => address,
+            UntranslatedEndpoint::Peer(PeerEndpoint { address, .. }) => address,
+        }
+    }
+}
+
+/// Data used to issue connections to a node.
+///
+/// Fetched from the cluster in Metadata.
+#[non_exhaustive] // <- so that we can add more fields in a backwards-compatible way
+#[derive(Clone)]
+pub struct PeerEndpoint {
+    pub host_id: Uuid,
+    pub address: SocketAddr,
+    pub datacenter: Option<String>,
+    pub rack: Option<String>,
+}
+
+impl From<Peer> for PeerEndpoint {
+    fn from(peer: Peer) -> Self {
+        Self {
+            host_id: peer.host_id,
+            address: peer.address,
+            datacenter: peer.datacenter,
+            rack: peer.rack,
+        }
+    }
+}
```

Comment:
Do we need to expose this conversion? Is it used anywhere? I'm not sure whether it is useful to convert a Peer by value into a PeerEndpoint.

I'd rather not introduce such impls unnecessarily as they are a part of the API.

Reply:
Right, it's currently unused.